src/hooks/useSession.ts

```diff
@@ -1,16 +1,32 @@
 import { useEffect, useState } from 'react';
 import { uuidv7 } from 'uuidv7';
 
-export const useSession = (): string => {
-  const sessionIdFromStorage = sessionStorage?.getItem('session_id');
-  const [session, setSession] = useState('');
+export const useSession = (): string | undefined => {
+  let sessionIdFromStorage: string | false | null = null; // Initialize with a default value
+
+  // Check if sessionStorage is available
+  if (typeof window !== 'undefined') {
+    sessionIdFromStorage =
+      typeof sessionStorage !== 'undefined' &&
+      sessionStorage.getItem('session_id');
+  }
+
+  const [session, setSession] = useState<string | undefined>('');
 
   useEffect(() => {
     if (sessionIdFromStorage) {
       setSession(sessionIdFromStorage);
     } else {
-      setSession(uuidv7());
-      sessionStorage?.setItem('session_id', session);
+      const newSessionId = uuidv7();
+      setSession(newSessionId);
+
+      // Ensure sessionStorage is available before setting item
+      if (
```

Comment:
Hum, seems to be a duplication of line 8 because you are already checking if it exists or not, better to go out of the hook if not exists, wondering if 'use client' can help or not